cranelift/codegen/src/isa/x86/binemit.rs

```diff
@@ -209,6 +220,150 @@ fn put_rexmp3<CS: CodeSink + ?Sized>(bits: u16, rex: u8, sink: &mut CS) {
     sink.put1(bits as u8);
 }
 
+/// Defines the EVEX context for the `L'`, `L`, and `b` bits (bits 6:4 of EVEX P2 byte). Table 2-36 in
+/// section 2.6.10 (Intel Software Development Manual, volume 2A) describes how these bits can be
+/// used together for certain classes of instructions; i.e., special should be taken to ensure
```

Comment:
Added this in a fixup and rebased so that the history looks cleaner; like we discussed, I think there are some bug fixes here that we won't want to squash together so I have been carefully rebasing.